tests/ui/cli/crate-type-flag.rs

```diff
@@ -0,0 +1,57 @@
+//! Check that `rustc`'s `--crate-type` flag accepts `--crate-type=<valid_type>` as well as the
+//! multi-value version `--crate-type=<valid_type_1>,<valid_type_2>`.
+//!
+//! This test does not try to check if the output artifacts are valid.
+
+// FIXME(#132309): add a proper `supports-crate-type` directive.
+
+// Single valid crate types should pass
+//@ revisions: lib_ rlib_ staticlib_ dylib_ cdylib_ bin_ proc_macro_
```

Comment:
I assume all these have an underscore suffix for consistency with the last case? May I suggest either:
- removing the `_` from the regular ones, and having `proc-macro` as `proc_dash_macro` (with symmetry with `proc_macro`)
- or, arguable worse, having "crate type" in the revision name, like for the empty crate type test

Reply:
I recall you cannot use not-underscored revision name, because they get registered as `cfg`s, but then IIRC you can't use those as cfg names. I'll have to double-check.

Reply:
That's only for `proc_macro` IIRC.

Reply:
EDIT: ah it's because `proc_macro` is a built-in cfg, I just appened suffix to be consistent, I could use the non-underscore version for non-proc-macro as you suggested.

cf. https://github.com/rust-lang/rust/pull/130599#discussion_r1768914554

Reply:
Went with

> removing the `_` from the regular ones, and having `proc-macro` as `proc_dash_macro` (with symmetry with `proc_macro`)